spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/store/JwtAccessTokenConverter.java

```diff
@@ -76,6 +76,8 @@
 	private String signingKey = verifierKey;
 
 	private SignatureVerifier verifier;
+	
+	private Map<String, String> headers = new LinkedHashMap<String, String>();
```

Comment:
`JwtAccessTokenConverter` is **not thread-safe** so `headers` attribute cannot be added here. If you still require the ability to add custom headers to the JWT then please re-work the solution and submit a new PR that includes tests.